developer-extension/src/panel/components/tabs/eventsTab/facetList.tsx

```diff
@@ -0,0 +1,170 @@
+import { BORDER_RADIUS, tabsListBorder } from '../../../uiUtils'
+
+export function FacetList({
+  facetRegistry,
+  excludedFacetValues,
+  onExcludedFacetValuesChange,
+}: {
+  facetRegistry: FacetRegistry
+  excludedFacetValues: ExcludedFacetValues
+  onExcludedFacetValuesChange: (newExcludedFacetValues: ExcludedFacetValues) => void
+}) {
+  return (
+    <FacetField
+      facet={FACETS[0]}
+      depth={0}
+      facetRegistry={facetRegistry}
+      excludedFacetValues={excludedFacetValues}
+      onExcludedFacetValuesChange={onExcludedFacetValuesChange}
+    />
+  )
+}
+
+function FacetField({
+  facet,
+  depth,
+  facetRegistry,
+  excludedFacetValues,
+  onExcludedFacetValuesChange,
+}: {
+  facet: Facet
+  depth: number
+  facetRegistry: FacetRegistry
+  excludedFacetValues: ExcludedFacetValues
+  onExcludedFacetValuesChange: (newExcludedFacetValues: ExcludedFacetValues) => void
+}) {
+  const facetValueCounts = facetRegistry.getFacetValueCounts(facet.id)
+
+  return (
+    <Box>
+      {facet.label && (
+        <Text pt="xs" color="dimmed" weight="bold">
+          {facet.label}
+        </Text>
+      )}
+
+      {Array.from(facetValueCounts.entries(), ([facetValue, facetValueCount]) => (
+        <FacetValue
+          key={facetValue}
+          facet={facet}
+          facetValue={facetValue}
+          facetValueCount={facetValueCount}
+          depth={depth}
+          facetRegistry={facetRegistry}
+          excludedFacetValues={excludedFacetValues}
+          onExcludedFacetValuesChange={onExcludedFacetValuesChange}
+        />
+      ))}
+    </Box>
+  )
+}
+
+const SPACE_BETWEEN_CHECKBOX = 4 // the smallest size defined by mantine ("xs") is 10px, which is a bit too much
+
+function FacetValue({
+  facet,
+  facetValue,
+  facetValueCount,
+  depth,
+  facetRegistry,
+  excludedFacetValues,
+  onExcludedFacetValuesChange,
+}: {
+  facet: Facet
+  facetValue: FacetValue
+  facetValueCount: number
+  depth: number
+  facetRegistry: FacetRegistry
+  excludedFacetValues: ExcludedFacetValues
+  onExcludedFacetValuesChange: (newExcludedFacetValues: ExcludedFacetValues) => void
+}) {
+  const isTopLevel = depth === 0
+  const isSelected = !excludedFacetValues[facet.id] || !excludedFacetValues[facet.id].includes(facetValue)
+  const value = (
+    <Flex justify="space-between" mt={isTopLevel ? 'xs' : SPACE_BETWEEN_CHECKBOX}>
+      <Checkbox
+        label={facet.values?.[facetValue]?.label ?? facetValue}
+        checked={isSelected}
+        onChange={() => {
+          onExcludedFacetValuesChange(toggleExcludedFacetValue(facet, excludedFacetValues, facetValue))
+        }}
+      />
+      <Text>{facetValueCount}</Text>
+    </Flex>
+  )
+
+  const childFacets = FACETS.filter((otherFacet) => otherFacet.parent === `${facet.id}:${facetValue}`)
+  const children = childFacets.length > 0 && (
+    <Collapse in={isSelected}>
+      <Box
+        ml={isTopLevel ? undefined : 9}
+        pl={isTopLevel ? undefined : 9}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why 9?